### PR TITLE
update build steps for linux

### PR DESCRIPTION
**Pull request by vhvb1989**
- Removing build-tools folder.  Adding direct `cmake` commands in Readme for linux.

### CODE REVIEW

**Comment by vhvb1989** [comment]
We need to update cmake to fail on Warnings on linux and Windows
Currently only Mac is failing do to warnings.
@barcharcraz , do you know how/where to add the compiler config within this project for `storage/ustorageclient/samples` ??  

I tried adding `target_compile_options(${this_sample_name} PRIVATE -Wall -Wextra)` to each sample's CMakeList.txt but Cmake didn't like it XD... wonder if you can help

**Comment by barcharcraz** [comment]
> We need to update cmake to fail on Warnings on linux and Windows
> Currently only Mac is failing do to warnings.
> @barcharcraz Charlie Barto FTE , do you know how/where to add the compiler config within this project for `storage/ustorageclient/samples` ??
> 
> I tried adding `target_compile_options(${this_sample_name} PRIVATE -Wall -Wextra)` to each sample's CMakeList.txt but Cmake didn't like it XD... wonder if you can help

I'm not sure why cmake doesn't like that, I would say add_compile_options is "fine" for this case, just make sure it's not unconditional.

File: README.md [hunk 1]
```diff
@@ -14,13 +14,25 @@ CMake version 3.12 is required to use these libraries.
...
+> Note: Project contains git submodules required to build. Before building run `git submodule update --init --recursive`
+
 #### Windows
 Use PowerShell to run {projectDir}/build.ps1
 
 #### Linux
-Use bash to run {projectDir}/build-tools/build.sh
-
-Output libraries are created in /build folder
+- Install `libssl`
```
Comment (barcharcraz) [inline README.md]: specify version please (or say it doesn't matter) not everyone is on ubuntu and this will get out of date over time.

I'd prefer saying "install openssl version .... On ubuntu XX.XX this is libssl-dev"

same for libuuid.

File: README.md [hunk 2]
```diff
@@ -14,13 +14,25 @@ CMake version 3.12 is required to use these libraries.
...
+  - ```sudo apt-get install uuid-dev```
+
+##### steps to build
+```
+mkdir build
+build
+cmake ../
+make
+```
+> Note: use `cmake -Duse_default_uuid=ON ../` if no uuid-dev is installed in system
```
Comment (barcharcraz) [inline README.md]: this embeds libuuid correct?

File: README.md [hunk 3]
```diff
@@ -14,13 +14,25 @@ CMake version 3.12 is required to use these libraries.
 ### Development Environment
 Project contains files to work on Windows or Linux based OS.
 
+> Note: Project contains git submodules required to build. Before building run `git submodule update --init --recursive`
```
Comment (barcharcraz) [inline README.md]: ideally outline which submodules are required here, and how one may use their own version instead of the bundled one (important for package maintainers)

File: sdk/storage/ustorageclient/samples/blob_append_bytes/main.c [hunk 4]
```diff
@@ -70,7 +70,7 @@ void main(void)
         (void)printf("Uploaded Data:\n%s\n", data);
         (void)printf("Downloaded Data:\n%.*s\n", (int)file_size, file);
         // compare
-        if(strncmp(data, file, sizeof(data)) == 0)
+        if(strncmp((const char *)data, (const char *)file, sizeof(data)) == 0)
```
Comment (barcharcraz) [inline sdk/storage/ustorageclient/samples/blob_append_bytes/main.c]: why is this needed?

File: sdk/storage/ustorageclient/samples/blob_append_bytes/main.c [hunk 5]
```diff
@@ -82,4 +82,6 @@ void main(void)
 
     // end sample
     (void)printf("\n********** Sample '%s' FINISH **********\r\n", AZSTORAGE_CONFIG_SAMPLE_NAME);
```
Comment (barcharcraz) [inline sdk/storage/ustorageclient/samples/blob_append_bytes/main.c]: what is the carrage-return doing here? printf will "do the right thing" on windows even without it? Did you reopen stdout in binary mode someplace?

File: README.md (hunk 2 above)
Comment (vhvb1989) [reply README.md]: makes compilation works when uuid is not in the system (like in DevOps Mac env).
Didn't run tests to see if the custom impl works thou

File: sdk/storage/ustorageclient/samples/blob_append_bytes/main.c (hunk 4 above)
Comment (vhvb1989) [reply sdk/storage/ustorageclient/samples/blob_append_bytes/main.c]: compilation fails on Mac if casting is not added:
```
[ 97%] Building C object sdk/storage/ustorageclient/samples/blob_append_bytes/CMakeFiles/blob_append_bytes_exe.dir/main.c.o
/Users/vsts/agent/2.155.1/work/1/s/sdk/storage/ustorageclient/samples/blob_append_bytes/main.c:73:20: error: passing 'const unsigned char [12]' to parameter of type 'const char *' converts between pointers to integer types with different sign [-Werror,-Wpointer-sign]
        if(strncmp(data, file, sizeof(data)) == 0)
                   ^~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/string.h:84:26: note: passing argument to parameter '__s1' here
int      strncmp(const char *__s1, const char *__s2, size_t __n);
                             ^
/Users/vsts/agent/2.155.1/work/1/s/sdk/storage/ustorageclient/samples/blob_append_bytes/main.c:73:26: error: passing 'unsigned char [1024]' to parameter of type 'const char *' converts between pointers to integer types with different sign [-Werror,-Wpointer-sign]
        if(strncmp(data, file, sizeof(data)) == 0)
                         ^~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/string.h:84:44: note: passing argument to parameter '__s2' here
int      strncmp(const char *__s1, const char *__s2, size_t __n);
                                               ^
2 errors generated.

```

I added it to make Mac pipeline works. And this is the reason I would like to have the same Warnings on Linux/Windows

File: README.md (hunk 1 above)
Comment (vhvb1989) [reply README.md]: makes sense.  Will update it. tks
